src/test/java/com/thirdcc/webapp/web/rest/EventAttendeeResourceIT.java

```diff
@@ -133,13 +219,133 @@ public void createEventAttendee() throws Exception {
         List<EventAttendee> eventAttendeeList = eventAttendeeRepository.findAll();
         assertThat(eventAttendeeList).hasSize(databaseSizeBeforeCreate + 1);
         EventAttendee testEventAttendee = eventAttendeeList.get(eventAttendeeList.size() - 1);
-        assertThat(testEventAttendee.getUserId()).isEqualTo(DEFAULT_USER_ID);
-        assertThat(testEventAttendee.getEvenId()).isEqualTo(DEFAULT_EVEN_ID);
+        assertThat(testEventAttendee.getUserId()).isEqualTo(user.getId());
+        assertThat(testEventAttendee.getEventId()).isEqualTo(savedEvent.getId());
         assertThat(testEventAttendee.isProvideTransport()).isEqualTo(DEFAULT_PROVIDE_TRANSPORT);
     }
 
     @Test
-    @Transactional
+    public void createEventAttendee_WithNonExistingUserId_ShouldThrow400() throws Exception {
+        Event savedEvent = initEventDB();
+        int databaseSizeBeforeCreate = eventAttendeeRepository.findAll().size();
+
+        // Create the EventAttendee
+        EventAttendeeDTO eventAttendeeDTO = createDefaultEventAttendeeDTO();
+        eventAttendeeDTO.setUserId(NON_EXIST_USER_ID);
+        eventAttendeeDTO.setEventId(savedEvent.getId());
+
+        restEventAttendeeMockMvc.perform(post("/api/event-attendees")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventAttendeeDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventAttendee> eventAttendeeList = eventAttendeeRepository.findAll();
+        assertThat(eventAttendeeList).hasSize(databaseSizeBeforeCreate);
+    }
+
+    @Test
+    public void createEventAttendee_WithExistedAttendee_ShouldThrow400() throws Exception {
+        EventAttendee eventAttendee = initEventAttendeeDB();
+        int databaseSizeBeforeCreate = eventAttendeeRepository.findAll().size();
+
+        // Create the EventAttendee
+        EventAttendeeDTO eventAttendeeDTO = createDefaultEventAttendeeDTO();
+        eventAttendeeDTO.setUserId(eventAttendee.getUserId());
+        eventAttendeeDTO.setEventId(eventAttendee.getEventId());
+
+        restEventAttendeeMockMvc.perform(post("/api/event-attendees")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventAttendeeDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventAttendee> eventAttendeeList = eventAttendeeRepository.findAll();
+        assertThat(eventAttendeeList).hasSize(databaseSizeBeforeCreate);
+    }
+
+    @Test
+    public void createEventAttendee_WithNonExistingEvent_ShouldThrow400() throws Exception {
+        int databaseSizeBeforeCreate = eventAttendeeRepository.findAll().size();
+
+        EventAttendeeDTO eventAttendeeDTO = createDefaultEventAttendeeDTO();
+        eventAttendeeDTO.setUserId(user.getId());
+        eventAttendeeDTO.setEventId(event.getId());
```

Comment:
can use `Long.MAX_VALUE` to recreate the scenario of a nonexisting event
in your code, `event.getId()` can be null or a valid eventId. who knows?
```
eventAttendeeDTO.setEventId(Long.MAX_VALUE);
```

Reply:
Solved in 5cde1adf7995840ccd786d96e4de3e4eddd348f5